crates/stc_ts_file_analyzer/src/analyzer/assign/mod.rs

```diff
@@ -3028,7 +3028,23 @@ impl Analyzer<'_, '_> {
 
                 return Ok(());
             }
-
+            Type::Param(param) => {
+                if let Some(constraint) = &param.constraint {
+                    if !constraint.is_str_like() && !constraint.is_never() {
+                        return Err(ErrorKind::NotSatisfyConstraint {
+                            span: param.span,
```

Comment:
span is wrong here, currently it's
```rs
 175 │ type X<T extends number> = Uppercase<T>;
     ·        ────────────────
```
but it needs to be
```rs
```rs
 175 │ type X<T extends number> = Uppercase<T>;
     ·                                      ─
```
looking into it
```

Reply:
Fixed but still slightly wrong, it's
```rs
 │ type X<T extends number> = Uppercase<T>;
     ·                                 ───
```
but it needs to be
```rs
 │ type X<T extends number> = Uppercase<T>;
     ·                                  ─
```
IMO it's kinda ok because string intrinsics only take one param.

Reply:
I agree with you. I don't think it matters for the initial version